### DISCUSSION TITLE
信息检索 - CS自学指南

### DISCUSSION THREAD

**Discussion by giscus** [PKUFlyingPig/cs-self-learning#370] [Announcements]
# 信息检索 - CS自学指南

CS自学指南

https://csdiy.wiki/%E5%BF%85%E5%AD%A6%E5%B7%A5%E5%85%B7/%E4%BF%A1%E6%81%AF%E6%A3%80%E7%B4%A2/

**Comment by JinBiLianShao** [comment]
Good

**Reply by wuqi** [reply]
其实还是搬瓦工机场稳定些,要钱但是不多...谷歌搜中文不咋地,ecosia或者bing靠谱一些,现在了就别迷信google了,体验现在还真没上来

**Comment by lijiaxin2333** [comment]
NICE!!!

**Comment by CainLin219** [comment]
非常感谢，令人受益匪浅，我也要开始我的CS61学习了~

**Comment by glitterandgloss** [comment]
计算机视觉推荐CS231n

**Comment by Github-632** [comment]
cool

**Comment by Github-632** [comment]
## cool

**Comment by LeeZhongshi** [comment]
佩服！

**Comment by 123jc-hu** [comment]
感谢！

**Comment by GoingBin** [comment]
开启这段奇妙的旅程吧

**Comment by current2020** [comment]
感谢！

**Comment by OrionHe** [comment]
感谢！

**Comment by Luc12138** [comment]
(https://www.bilibili.com/video/BV1w54y1q7uf/?spm_id_from=333.337.search-card.all.click&vd_source=3c2707cbc6a9ebd48fd79fce2071bbf7#t=201.645433)
如何精准搜索

**Comment by setarcox** [comment]
目前信息检索的方式基本可以完全通过chatgpt替代

**Reply by wuqi** [reply]
其实粗略信息推荐,比如入门啊什么的推荐chatGPT,细节的话,老老实实读文档...大部分时候都是错的,给的很多网址现在还是不对,比如你要搜具体的shader效果什么的,会给shadertoy的示例网站但是完全打不开...

针对具体库的问题大概有五到七成概率错的或者是废话,还不如直接搜stackoverflow...

**Reply by kechenkristin** [reply]
Agree

**Comment by KinGeorge** [comment]
现在需要更新chatgpt的信息检索了（虽然好像挺简单的

**Reply by wuqi** [reply]
写了但是没得时间改,娃半岁晚上压根没时间,上班摸鱼不能摸这么大

**Comment by DentistCode** [comment]
感谢，这些基础的搜索原理还是需要掌握的。
但是可以尝试一下new bing，真的很好用: P

**Comment by Michael18811380328** [comment]
nice

**Comment by Donga5** [comment]
信息检索的话，感觉现在这个时代，除了goolge以外，不得不提到的东西提示是大模型。编写好的prompt让大模型产生更好的输出结果也是一个不错的选择。

**Comment by Donga5** [comment]
1.另外，大家需要魔法的。可以尝试申请**github student pack**。这个学生包中提供Azure一年100每年的额度。然后利用这100美元购买一台国外的vps，自建节点，速度一个人用的真的飞快。
2.难度的话，相信对于看到这篇文章的大家来说应该不是什么问题。

**Reply by Andrew-moon** [reply]
赞同，而且GitHub学生包还包含GitHub copilot，是用的gpt4，可以辅助写代码，非常强。似乎GitHub上还有通过获取copilot的api（或者是token？）来白嫖gpt4的，不过我最近搜发现这个库已经被封了😭
另外云服务器的话vultr也可以试试，按小时计费，用完就可以释放掉了

**Comment by dghtucs** [comment]
建了一个关于CS必学工具课程QQ群，想讨论可入746305701

**Comment by Himalian** [comment]
ripgrep ，挺不错的一个本地代码搜索工具，基于命令行

**Comment by llzzzzoo** [comment]
佬什么时候更新一下如何更好的用 LLM 来检索信息呀，期待ing

**Reply by wuqi** [reply]
https://github.com/wangshusen/SearchEngine  你可以直接看看这个,从原理入手,其实ai检索跟普通的搜索引擎区别不大,不过注意幻觉.
在一个就是,描述具体的问题,如果你有思路,把思路分步写进去,参照提问的智慧里那种提问方式,会获取到更好的信息,举个例子:

`我有一些std::vector<ogrFeature> featureList,都是线要素  我希望通过geos抽稀化简输出到 std::vector<ogrFeature>outList中,具体步骤如下:
1. 将所有线进行dissolve
2. 设定容差 ,进行snappingnode,将其合并打断
3.遍历打断的geometry,用bounding查找到第一个要素赋值属性给它,新建一个feature,输入到outlist里`

这种基本能给出一个不能跑但是思路大致正确的版本,直接要求给出一个gdal化简的示例基本给不出啥像样的例子

**Comment by Haining-Li** [comment]
no, searchcode 入不敷出关闭了

**Comment by lil-Journey** [comment]
自建日本节点服务器 月租5元不限流量 本人也在用 失效时可联系人工修复 qq:1165116287
